<div style="border:solid blue 2px; padding: 20px">
  
**Hello Rossi.**

My name is Yulia, and I will be reviewing your project. 

You will find my comments in coloured cells marked as 'Reviewer's comment'. The cell colour will vary based on the contents - I am explaining it further below. 

**Note:** Please do not remove or change my comments - they will help me in my future reviews and will make the process smoother for both of us. 

<div class="alert alert-success"; style="border-left: 7px solid green">
<b>✅ Reviewer's comment</b> 
    
Such comment will mark efficient solutions and good ideas that can be used in other projects.
</div>

<div class="alert alert-warning"; style="border-left: 7px solid gold">
<b>⚠️ Reviewer's comment</b> 
    
The parts marked with yellow comments indicate that there is room for optimisation. Though the correction is not necessary it is good if you implement it.
</div>

<div class="alert alert-danger"; style="border-left: 7px solid red">
<b>⛔️ Reviewer's comment</b> 
    
If you see such a comment, it means that there is a problem that needs to be fixed. Please note that I won't be able to accept your project until the issue is resolved.
</div>

You are also very welcome to leave your comments / describe the corrections you've done / ask me questions, marking them with a different colour. You can use the example below: 

<div class="alert alert-info"; style="border-left: 7px solid blue">
<b>Student's comment</b>

<div style="border:solid green 2px; padding: 20px">
    
<div class="alert alert-success">
<b>Review summary</b> 
    
Rossi, thanks for submitting the project. You did a great job and it was a pleasure to review your project.
    
- I can see that you already know many python methods that allow to thoroughly study dataframes 
- You know how to find and remove missing values and duplicates
- You can write your own functions, use conditional indexing and many other python methods
    
<div class="alert alert-danger">

However, there are a few critical comments that need to be corrected. You will find them in the red-coloured cells in relevant sections.
    
Could you please fix them and send the project back to me?


</div>
    
You may also work on the yellow comments. If you have any questions please write them when you return your project. 
    
I'll be looking forward to getting your updated notebook.

## Basic Python - Project <a id='intro'></a>

## Introduction <a id='intro'></a>
In this project, you will work with data from the entertainment industry. You will study a dataset with records on movies and shows. The research will focus on the "Golden Age" of television, which began in 1999 with the release of *The Sopranos* and is still ongoing.

The aim of this project is to investigate how the number of votes a title receives impacts its ratings. The assumption is that highly-rated shows (we will focus on TV shows, ignoring movies) released during the "Golden Age" of television also have the most votes.

<div class="alert alert-success"; style="border-left: 7px solid green">
<b>✅ Reviewer's comment, v. 1</b> 
    
Title and introduction are essential parts of the project. Make sure you do not forget to include it in your further projects. 
    
It is optimal if introduction consists of:
    
- brief description of the situation;
- goal of the project;
- description of the data we are going to use.
</div>


### Stages 
Data on movies and shows is stored in the `/datasets/movies_and_shows.csv` file. There is no information about the quality of the data, so you will need to explore it before doing the analysis.

First, you'll evaluate the quality of the data and see whether its issues are significant. Then, during data preprocessing, you will try to account for the most critical problems.
 
Your project will consist of three stages:
 1. Data overview
 2. Data preprocessing
 3. Data analysis

## Stage 1. Data overview <a id='data_review'></a>

Open and explore the data.

You'll need `pandas`, so import it.

In [1]:
import pandas as pd


<div class="alert alert-success"; style="border-left: 7px solid green">
<b>✅ Reviewer's comment, v. 1</b> 
    

    
It's a good practice to upload all necessary libraries in the first code cell, separately from datasets (so that you do not have to reload the dataset if you need to add another library). 

Read the `movies_and_shows.csv` file from the `datasets` folder and save it in the `df` variable:

In [2]:
df = pd.read_csv("/datasets/movies_and_shows.csv")


Print the first 10 table rows:

In [3]:
df.head(10)


name                      Character   r0le        TITLE   Type  \
0   Robert De Niro                  Travis Bickle  ACTOR  Taxi Driver  MOVIE   
1     Jodie Foster                  Iris Steensma  ACTOR  Taxi Driver  MOVIE   
2    Albert Brooks                            Tom  ACTOR  Taxi Driver  MOVIE   
3    Harvey Keitel        Matthew 'Sport' Higgins  ACTOR  Taxi Driver  MOVIE   
4  Cybill Shepherd                          Betsy  ACTOR  Taxi Driver  MOVIE   
5      Peter Boyle                         Wizard  ACTOR  Taxi Driver  MOVIE   
6   Leonard Harris      Senator Charles Palantine  ACTOR  Taxi Driver  MOVIE   
7   Diahnne Abbott                Concession Girl  ACTOR  Taxi Driver  MOVIE   
8      Gino Ardito             Policeman at Rally  ACTOR  Taxi Driver  MOVIE   
9  Martin Scorsese  Passenger Watching Silhouette  ACTOR  Taxi Driver  MOVIE   

   release Year              genres  imdb sc0re  imdb v0tes  
0          1976  ['drama', 'crime']         8.2    808582.0  
1          1976  ['drama', 'crime']         8.2    808582.0  
2          1976  ['drama', 'crime']         8.2    808582.0  
3          1976  ['drama', 'crime']         8.2    808582.0  
4          1976  ['drama', 'crime']         8.2    808582.0  
5          1976  ['drama', 'crime']         8.2    808582.0  
6          1976  ['drama', 'crime']         8.2    808582.0  
7          1976  ['drama', 'crime']         8.2    808582.0  
8          1976  ['drama', 'crime']         8.2    808582.0  
9          1976  ['drama', 'crime']         8.2    808582.0

Obtain the general information about the table with one command:

In [4]:
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 85579 entries, 0 to 85578
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0      name       85579 non-null  object 
 1   Character     85579 non-null  object 
 2   r0le          85579 non-null  object 
 3   TITLE         85578 non-null  object 
 4     Type        85579 non-null  object 
 5   release Year  85579 non-null  int64  
 6   genres        85579 non-null  object 
 7   imdb sc0re    80970 non-null  float64
 8   imdb v0tes    80853 non-null  float64
dtypes: float64(2), int64(1), object(6)
memory usage: 5.9+ MB


<div class="alert alert-success"; style="border-left: 7px solid green">
<b>✅ Reviewer's comment, v. 1</b> 
    
Great - you've used a comprehensive set of methods to have a first look at the data.

The table contains nine columns. The majority store the same data type: object. The only exceptions are `'release Year'` (int64 type), `'imdb sc0re'` (float64 type) and `'imdb v0tes'` (float64 type). Scores and votes will be used in our analysis, so it's important to verify that they are present in the dataframe in the appropriate numeric format. Three columns (`'TITLE'`, `'imdb sc0re'` and `'imdb v0tes'`) have missing values.

According to the documentation:
- `'name'` — actor/director's name and last name
- `'Character'` — character played (for actors)
- `'r0le '` — the person's contribution to the title (it can be in the capacity of either actor or director)
- `'TITLE '` — title of the movie (show)
- `'  Type'` — show or movie
- `'release Year'` — year when movie (show) was released
- `'genres'` — list of genres under which the movie (show) falls
- `'imdb sc0re'` — score on IMDb
- `'imdb v0tes'` — votes on IMDb

We can see three issues with the column names:
1. Some names are uppercase, while others are lowercase.
2. There are names containing whitespace.
3. A few column names have digit '0' instead of letter 'o'. 


<div class="alert alert-success"; style="border-left: 7px solid green">
<b>✅ Reviewer's comment, v. 1</b> 
    


Please note that in the next python projects you need to comment your actions, write your observations and insights. 
    
You need to use Markdown type of cells for that. To change the cell type for Markdown please use the upper navigation bar - create a new cell with `+` button, step on it and change Raw to Markdown in the dropdown menu.

### Conclusions <a id='data_review_conclusions'></a> 

Each row in the table stores data about a movie or show. The columns can be divided into two categories: the first is about the roles held by different people who worked on the movie or show (role, name of the actor or director, and character if the row is about an actor); the second category is information about the movie or show itself (title, release year, genre, imdb figures).

It's clear that there is sufficient data to do the analysis and evaluate our assumption. However, to move forward, we need to preprocess the data.

<div class="alert alert-success"; style="border-left: 7px solid green">
<b>✅ Reviewer's comment, v. 1</b> 
    
Please note that it is highly recommended to add a conclusion / summary after each section and describe briefly your observations and / or major outcomes of the analysis.

## Stage 2. Data preprocessing <a id='data_preprocessing'></a>
Correct the formatting in the column headers and deal with the missing values. Then, check whether there are duplicates in the data.

In [5]:
print("Column names in the DataFrame:")
print(df.columns.tolist())

Column names in the DataFrame:
['   name', 'Character', 'r0le', 'TITLE', '  Type', 'release Year', 'genres', 'imdb sc0re', 'imdb v0tes']


Change the column names according to the rules of good style:
* If the name has several words, use snake_case
* All characters must be lowercase
* Remove whitespace
* Replace zero with letter 'o'

In [6]:
df = df.rename(columns={'   name': 'name', 'Character':'character', 'r0le': 'role', '  Type': 'type', 'TITLE':'title', 'release Year':'release_year', 'imdb sc0re':'imdb_score', 'imdb v0tes':'imdb_votes'} )
    



<div class="alert alert-success"; style="border-left: 7px solid green">
<b>✅ Reviewer's comment, v. 1</b> 
    
This is a good way to rename the columns. And you've set proper names, in line with Python naming convention.

Check the result. Print the names of the columns once more:

In [7]:
print(df.columns.tolist())


['name', 'character', 'role', 'title', 'type', 'release_year', 'genres', 'imdb_score', 'imdb_votes']


### Missing values <a id='missing_values'></a>
First, find the number of missing values in the table. To do so, combine two `pandas` methods:

In [8]:
missing_values_count = df.isnull().sum()
print(missing_values_count)

name               0
character          0
role               0
title              1
type               0
release_year       0
genres             0
imdb_score      4609
imdb_votes      4726
dtype: int64


Not all missing values affect the research: the single missing value in `'title'` is not critical. The missing values in columns `'imdb_score'` and `'imdb_votes'` represent around 6% of all records (4,609 and 4,726, respectively, of the total 85,579). This could potentially affect our research. To avoid this issue, we will drop rows with missing values in the `'imdb_score'` and `'imdb_votes'` columns.

In [9]:
df.dropna(subset=['title', 'imdb_score', 'imdb_votes'], inplace=True)


Make sure the table doesn't contain any more missing values. Count the missing values again.

In [10]:
missing_values_count = df.isnull().sum()
print(missing_values_count)


name            0
character       0
role            0
title           0
type            0
release_year    0
genres          0
imdb_score      0
imdb_votes      0
dtype: int64


<div class="alert alert-success"; style="border-left: 7px solid green">
<b>✅ Reviewer's comment, v. 1</b> 
    
Well done, you've found and removed all the missing values.

### Duplicates <a id='duplicates'></a>
Find the number of duplicate rows in the table using one command:

In [11]:
duplicate_rows = df.duplicated().sum()
print("Number of duplicate rows:", duplicate_rows)


Number of duplicate rows: 6994


Review the duplicate rows to determine if removing them would distort our dataset.

In [12]:
duplicate_rows = df[df.duplicated(keep=False)]
print("Duplicate Rows:")
print(duplicate_rows)

Duplicate Rows:
                      name                  character      role  \
7560         Philip Greene  Baseball Fan (uncredited)     ACTOR   
7561         Philip Greene  Baseball Fan (uncredited)     ACTOR   
14502             Dan Levy                   Reporter     ACTOR   
14512             Dan Levy                   Reporter     ACTOR   
18951     Nicolas Le Nev??                    unknown  DIRECTOR   
...                    ...                        ...       ...   
85569       Jessica Cediel            Liliana Navarro     ACTOR   
85570  Javier Gardeaz?­bal   Agust??n "Peluca" Ort??z     ACTOR   
85571        Carla Giraldo             Valery Reinoso     ACTOR   
85572  Ana Mar??a S?­nchez                    Lourdes     ACTOR   
85577         Isabel Gaona                     Cacica     ACTOR   

                                 title   type  release_year  \
7560                   How Do You Know  MOVIE          2010   
7561                   How Do You Know  MOVIE        

<div class="alert alert-warning"; style="border-left: 7px solid gold">
<b>⚠️ Reviewer's comment, v. 1</b> 

Some dataframes can be quite big and reading the whole dataframe to display it might take significant resources. 
    
A more optimal way to check the contents of dataframes is using `head()`, `tail()` or `sample()` method. The first will display 5 first lines, the second one - 5 last lines, and `sample()` - a random line. You may specify a different number of lines you want to be displayed in the parentheses.

There are two clear duplicates in the printed rows. We can safely remove them.
Call the `pandas` method for getting rid of duplicate rows:

In [13]:
df.drop_duplicates(inplace=True)


Check for duplicate rows once more to make sure you have removed all of them:

In [14]:
print(df.duplicated().sum())


0


<div class="alert alert-success"; style="border-left: 7px solid green">
<b>✅ Reviewer's comment, v. 1</b> 
    
Well done.

Now get rid of implicit duplicates in the `'type'` column. For example, the string `'SHOW'` can be written in different ways. These kinds of errors will also affect the result.

Print a list of unique `'type'` names, sorted in alphabetical order. To do so:
* Retrieve the intended dataframe column 
* Apply a sorting method to it
* For the sorted column, call the method that will return all unique column values

In [15]:
types_column = df['type']
sorted_types = types_column.sort_values()
unique_sorted_types = sorted_types.unique()
print("Unique 'type' names sorted in alphabetical order:")
for t in unique_sorted_types:
    print(t)

Unique 'type' names sorted in alphabetical order:
MOVIE
SHOW
movies
shows
the movie
tv
tv series
tv show
tv shows


Look through the list to find implicit duplicates of `'show'` (`'movie'` duplicates will be ignored since the assumption is about shows). These could be names written incorrectly or alternative names of the same genre.

You will see the following implicit duplicates:
* `'shows'`
* `'SHOW'`
* `'tv show'`
* `'tv shows'`
* `'tv series'`
* `'tv'`

To get rid of them, declare the function `replace_wrong_show()` with two parameters: 
* `wrong_shows_list=` — the list of duplicates
* `correct_show=` — the string with the correct value

The function should correct the names in the `'type'` column from the `df` table (i.e., replace each value from the `wrong_shows_list` list with the value in `correct_show`).

In [16]:
def replace_wrong_show(wrong_shows_list, correct_show):
    for wrong_show in wrong_shows_list:
        df['type'] = df['type'].replace(wrong_show, correct_show)


Call `replace_wrong_show()` and pass it arguments so that it clears implicit duplicates and replaces them with `SHOW`:

In [17]:
replace_wrong_show(['shows', 'tv show', 'tv shows', 'tv series', 'tv'], 'SHOW')


<div class="alert alert-success"; style="border-left: 7px solid green">
<b>✅ Reviewer's comment, v. 1</b> 
    
You wrote an excellent function and removed the implicit duplicates.

Make sure the duplicate names are removed. Print the list of unique values from the `'type'` column:

In [18]:
unique_types = df['type'].sort_values().unique()
print(unique_types)


['MOVIE' 'SHOW' 'movies' 'the movie']


### Conclusions <a id='data_preprocessing_conclusions'></a>
We detected three issues with the data:

- Incorrect header styles
- Missing values
- Duplicate rows and implicit duplicates

The headers have been cleaned up to make processing the table simpler.

All rows with missing values have been removed. 

The absence of duplicates will make the results more precise and easier to understand.

Now we can move on to our analysis of the prepared data.

## Stage 3. Data analysis <a id='hypotheses'></a>

Based on the previous project stages, you can now define how the assumption will be checked. Calculate the average amount of votes for each score (this data is available in the `imdb_score` and `imdb_votes` columns), and then check how these averages relate to each other. If the averages for shows with the highest scores are bigger than those for shows with lower scores, the assumption appears to be true.

Based on this, complete the following steps:

- Filter the dataframe to only include shows released in 1999 or later.
- Group scores into buckets by rounding the values of the appropriate column (a set of 1-10 integers will help us make the outcome of our calculations more evident without damaging the quality of our research).
- Identify outliers among scores based on their number of votes, and exclude scores with few votes.
- Calculate the average votes for each score and check whether the assumption matches the results.

To filter the dataframe and only include shows released in 1999 or later, you will take two steps. First, keep only titles published in 1999 or later in our dataframe. Then, filter the table to only contain shows (movies will be removed).

In [19]:
# using conditional indexing modify df so it has only titles released after 1999 (with 1999 included)
# give the slice of dataframe new name
df_1999_or_later = df[df['release_year'] >= 1999]
shows_1999_or_later = df_1999_or_later[df_1999_or_later['type'] == 'SHOW']
print(shows_1999_or_later)

                      name                character      role      title  \
1664           Jeff Probst           Himself - Host     ACTOR   Survivor   
2076         Mayumi Tanaka  Monkey D. Luffy (voice)     ACTOR  One Piece   
2077          Kazuya Nakai     Roronoa Zoro (voice)     ACTOR  One Piece   
2078         Akemi Okamura             Nami (voice)     ACTOR  One Piece   
2079      Kappei Yamaguchi            Usopp (voice)     ACTOR  One Piece   
...                    ...                      ...       ...        ...   
85433     Maneerat Kam-Uan                       Ae     ACTOR  Let's Eat   
85434    Rudklao Amratisha                  unknown     ACTOR  Let's Eat   
85435      Jaturong Mokjok                  unknown     ACTOR  Let's Eat   
85436     Pisamai Wilaisak                  unknown     ACTOR  Let's Eat   
85437  Sarawut Wichiensarn                  unknown  DIRECTOR  Let's Eat   

       type  release_year                                             genres  \
1664   

<div class="alert alert-success"; style="border-left: 7px solid green">
<b>✅ Reviewer's comment, v. 1</b> 
    
Excellent application of conditional indexing.

<div class="alert alert-warning"; style="border-left: 7px solid gold">
<b>⚠️ Reviewer's comment, v. 1</b> 

The `print()` function is not good for displaying dataframes in Jupyter notebook. 
    
Sometimes you don't need any function to display a dataframe, for example, if you are calling it in the last line of a code cell.
    
In other cases it is preferrable to use `display()` instead of `print()`. Then dataframe will be shown in a table format, which is much easier to read.

In [20]:
# repeat conditional indexing so df has only shows (movies are removed as result)
shows_only = df[df['type'] == 'SHOW']
print(shows_only)

                      name             character      role  \
163         Graham Chapman               Various     ACTOR   
164          Michael Palin  Various / "It's" man     ACTOR   
165            Terry Jones               Various     ACTOR   
166              Eric Idle               Various     ACTOR   
167          Terry Gilliam               Various     ACTOR   
...                    ...                   ...       ...   
85433     Maneerat Kam-Uan                    Ae     ACTOR   
85434    Rudklao Amratisha               unknown     ACTOR   
85435      Jaturong Mokjok               unknown     ACTOR   
85436     Pisamai Wilaisak               unknown     ACTOR   
85437  Sarawut Wichiensarn               unknown  DIRECTOR   

                              title  type  release_year  \
163    Monty Python's Flying Circus  SHOW          1969   
164    Monty Python's Flying Circus  SHOW          1969   
165    Monty Python's Flying Circus  SHOW          1969   
166    Monty Python

<div class="alert alert-warning"; style="border-left: 7px solid gold">
<b>⚠️ Reviewer's comment, v. 1</b> 

You don't have to filter the dataframe for SHOWs again, as you already did it in the previous code cell. (The cell was here because it was expected that in the previous cell you filter dataframe for time period and in the above cell to filter for the certain type.)

The scores that are to be grouped should be rounded. For instance, titles with scores like 7.8, 8.1, and 8.3 will all be placed in the same bucket with a score of 8.

In [21]:
# rounding column with scores
df['imdb_score'] = df['imdb_score'].round(2)
#checking the outcome with tail()
print(df.tail())

                     name     character      role    title       type  \
85573       A??da Morales       Maritza     ACTOR  Lokillo  the movie   
85574    Adelaida Buscato    Mar??a Paz     ACTOR  Lokillo  the movie   
85575  Luz Stella Luengas  Karen Bayona     ACTOR  Lokillo  the movie   
85576        In??s Prieto         Fanny     ACTOR  Lokillo  the movie   
85578      Julian Gaviria       unknown  DIRECTOR  Lokillo  the movie   

       release_year      genres  imdb_score  imdb_votes  
85573          2021  ['comedy']         3.8        68.0  
85574          2021  ['comedy']         3.8        68.0  
85575          2021  ['comedy']         3.8        68.0  
85576          2021  ['comedy']         3.8        68.0  
85578          2021  ['comedy']         3.8        68.0  


<div class="alert alert-danger"; style="border-left: 7px solid red">
<b>⛔️ Reviewer's comment, v. 1</b> 

Please round the scores here to integer numbers, with 0 decimals.
    
The results below will change after you've changed the rounding. I'll review the rest later.

It is now time to identify outliers based on the number of votes.

In [22]:
# Use groupby() for scores and count all unique values in each group, print the result
score_counts = df.groupby('imdb_score').size()
print(score_counts)

imdb_score
1.5    10
1.6     9
1.7    14
1.9    12
2.0    27
       ..
9.0    38
9.1    17
9.2    19
9.3    12
9.5     8
Length: 78, dtype: int64


Based on the aggregation performed, it is evident that scores 2 (24 voted shows), 3 (27 voted shows), and 10 (only 8 voted shows) are outliers. There isn't enough data for these scores for the average number of votes to be meaningful.

To obtain the mean numbers of votes for the selected scores (we identified a range of 4-9 as acceptable), use conditional filtering and grouping.

In [23]:
# filter dataframe using two conditions (scores to be in the range 4-9)
filtered_df = df[(df['imdb_score'] >= 4) & (df['imdb_score'] <= 9)]
# group scores and corresponding average number of votes, reset index and print the result
score_avg_votes = filtered_df.groupby('imdb_score')['imdb_votes'].mean().reset_index()
print(score_avg_votes)

    imdb_score    imdb_votes
0          4.0  1.083360e+04
1          4.1  3.974943e+03
2          4.2  3.435804e+03
3          4.3  1.224099e+04
4          4.4  6.517818e+03
5          4.5  1.259273e+04
6          4.6  5.528028e+03
7          4.7  6.508528e+03
8          4.8  1.672401e+04
9          4.9  6.278013e+03
10         5.0  7.396346e+03
11         5.1  2.717094e+04
12         5.2  1.447588e+04
13         5.3  1.002977e+04
14         5.4  1.845894e+04
15         5.5  1.468271e+04
16         5.6  1.501672e+04
17         5.7  2.650358e+04
18         5.8  1.169624e+04
19         5.9  1.795570e+04
20         6.0  2.509318e+04
21         6.1  3.504980e+04
22         6.2  2.840788e+04
23         6.3  2.968597e+04
24         6.4  3.822340e+04
25         6.5  5.779703e+04
26         6.6  3.485852e+04
27         6.7  3.555035e+04
28         6.8  5.539657e+04
29         6.9  4.207586e+04
30         7.0  4.290969e+04
31         7.1  6.408959e+04
32         7.2  7.010848e+04
33         7.3

Now for the final step! Round the column with the averages, rename both columns, and print the dataframe in descending order.

In [24]:
# round column with averages
score_avg_votes['imdb_votes'] = score_avg_votes['imdb_votes'].round()
# rename columns
score_avg_votes.columns = ['imdb_score', 'average_votes']
# print dataframe in descending order
print(score_avg_votes.sort_values(by='average_votes', ascending=False))

    imdb_score  average_votes
48         8.8      1141579.0
47         8.7       577386.0
44         8.4       540650.0
45         8.5       411751.0
46         8.6       317712.0
43         8.3       185755.0
40         8.0       183888.0
41         8.1       183676.0
39         7.9       177880.0
50         9.0       153710.0
38         7.8       130218.0
42         8.2       116389.0
37         7.7       109220.0
35         7.5        88456.0
33         7.3        72266.0
49         8.9        70728.0
32         7.2        70108.0
34         7.4        69716.0
36         7.6        68840.0
31         7.1        64090.0
25         6.5        57797.0
28         6.8        55397.0
30         7.0        42910.0
29         6.9        42076.0
24         6.4        38223.0
27         6.7        35550.0
21         6.1        35050.0
26         6.6        34859.0
23         6.3        29686.0
22         6.2        28408.0
11         5.1        27171.0
17         5.7        26504.0
20        

The assumption macthes the analysis: the shows with the top 3 scores have the most amounts of votes.

## Conclusion <a id='hypotheses'></a>

The research done confirms that highly-rated shows released during the "Golden Age" of television also have the most votes. While shows with score 4 have more votes than ones with scores 5 and 6, the top three (scores 7-9) have the largest number. The data studied represents around 94% of the original set, so we can be confident in our findings.

<div class="alert alert-success"; style="border-left: 7px solid green">
<b>✅ Reviewer's comment, v. 1</b> 
    
Overall conclusion is an important part, where we should include the summary of the outcomes of the project.
    
Please note that you will need to write one for every project you make.